app/src/main/java/com/example/mystore/ui/screens/LoginScreen.kt

```diff
@@ -129,21 +142,49 @@ fun LoginScreen(navigateToProductDetailsScreen: () -> Unit = {}) {
             modifier = Modifier.fillMaxWidth(),
             onClick = {
                 isValidInput = validateFields(emailField, passwordField)
-                if (isValidInput && emailField == rightEmail && passwordField == rightPassword) {
-                    navigateToProductDetailsScreen()
+                if (isValidInput) {
+                    isButtonEnabled = false
+                    coroutineScope.launch(Dispatchers.Main) {
```

Comment:
For networking/IO prefer the `Dispatchers.IO` one. If you want you can read more about the differences [here](https://developer.android.com/kotlin/coroutines/coroutines-adv#main-safety), but generally prefer this one for non-UI related tasks, and keep in mind that there are others in general.